fendermint/vm/topdown/src/vote/operation/active.rs

```diff
@@ -35,15 +34,19 @@ impl ActiveOperationMode {
     }
 }
 
-#[async_trait]
-impl OperationModeHandler for ActiveOperationMode {
-    async fn advance(mut self) -> OperationStateMachine {
+impl<
+        T: GossipSender + Send + Sync + 'static + Clone,
+        S: GossipReceiver + Send + Sync + 'static,
+        V: VoteStore + Send + Sync + 'static,
+    > ActiveOperationMode<T, S, V>
+{
+    pub(crate) async fn advance(mut self) -> OperationStateMachine<T, S, V> {
         loop {
             select! {
                 Some(req) = self.handler.req_rx.recv() => {
-                    self.handler.handle_request(req);
+                    self.handler.handle_request(req, &self.metrics);
                 },
-                Ok(vote) = self.handler.gossip_rx.recv() => {
+                Ok(vote) = self.handler.gossip_rx.recv_vote() => {
```

Comment:
Where does the `Vote::v1_checked` validation is taken place?

Reply:
I delegated to `gossip`, maybe it's a good idea to handle it here.